SEED-platform/seed
# Salesforce scheduling issues

#### What's this PR do?
- Fixes an issue where attempting to set a Salesforce sync time with `hour = 0` or `minute = 0` would (in most cases) not actually take effect
- Fixes an issue where each time a Salesforce config is saved with a cron schedule that hasn't previously been used it orphans the old schedule, resulting in tons of unused records in the `crontabschedule` table:
![image](https://github.com/SEED-platform/seed/assets/411466/8889d322-bafa-4ecf-b5f3-74bdb026bc94)
- Improves the UI to use restricted numeric inputs for hours/minutes, which fixes mixed data types when saving a Salesforce config:
![image](https://github.com/SEED-platform/seed/assets/411466/379e84f8-1127-4d5a-810b-2b9498406840)

#### How should this be manually tested?
1. Modify the hour/minute of a Salesforce scheduled update and:
    1. Check that the `PUT` request uses integers for `update_at_hour` and `update_at_minute`
    2. Check that setting the hour or minute to `0` correctly updates the `crontabschedule` table
    3. Check that the `crontabschedule` table has no entries that aren't referenced by the `periodictask` table
